include/os/linux/spl/sys/kmem_cache.h

```diff
@@ -201,6 +201,12 @@ extern uint64_t spl_kmem_cache_entry_size(kmem_cache_t *cache);
     spl_kmem_cache_create(name, size, align, ctor, dtor, rclm, priv, vmp, fl)
 #define	kmem_cache_set_move(skc, move)	spl_kmem_cache_set_move(skc, move)
 #define	kmem_cache_destroy(skc)		spl_kmem_cache_destroy(skc)
+/*
+ * This is necessary to be compatible with other kernel modules
+ * or in-tree filesystem that may define kmem_cache_alloc,
+ * like bcachefs does it now.
+ */
+#undef kmem_cache_alloc
```

Comment:
I'd recommend wrapping this `#undef` with a `#ifdef kmem_cache_alloc` guard.  This will avoid any pedantic compiler warnings about undef'ing a define which doesn't exist.

Reply:
Good point!

Reply:
Done in https://github.com/openzfs/zfs/pull/15144 which is the rebased version.

Reply:
We can close this one if you want.

Reply:
Great.  Let's go ahead and open new PRs for the backports after it's merged.